binderhub/repoproviders.py

```diff
@@ -111,6 +121,21 @@ def has_higher_quota(self):
                 return True
         return False
 
+    def spec_configuration_override(self):
```

Comment:
What do you think of naming this `spec_config()` and taking care of finding and populating all the default values in this function? Then in the builder we wouldn't need to have any checks for "do we override this or not, what is the default". We have one place that is already ready to use.

The reason to call it `spec_config` is that it would be for one particular spec and not a whole bunch any more. Maybe we could even call it `repo_config`? Depends a bit on what words the builder uses (does it refer to building a spec or a repo). But this might be getting a bit OCD on naming :)

TL;DR: should we compute the configuration (combining over rides and defaults) for the spec here?

Reply:
I like it - that seems reasonable to me, lemme give it a stab and see how it looks

Reply:
see the latest push for an attempt at what this would look like, is this the kinda thing you had in mind? 

https://github.com/jupyterhub/binderhub/pull/888/commits/1399d58986da19c77139b8a9b8cedadc25c13c91#diff-c5688934f1e6dc3e932b6c84c1bbbd5dR133

In this case, `spec_config` would be defined at the helm chart config level, and repo_config is a method of RepoProvider that returns a dictionary of configuration values for that *repository* (which might have been updated from the spec_config setting)